posthog/api/dashboards/dashboard_tiles.py

```diff
@@ -0,0 +1,165 @@
+            dashboard_id=serializer.validated_data["dashboard_id"],
+            user=request.user,  # type: ignore
+            # only need to flag that a dashboard was added
+            before=lambda x: [],
+            after=lambda tile: [{"dashboard": {"id": tile.dashboard.id, "name": tile.dashboard.name}}],
+        )
+
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    @action(methods=["POST"], detail=False)
+    def remove(self, request: Request, *args: Any, **kwargs: Any) -> Response:
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        self.get_queryset().filter(**serializer.validated_data).update(deleted=True)
+
+        self._write_to_activity_log(
+            insight_id=serializer.validated_data["insight_id"],
+            dashboard_id=serializer.validated_data["dashboard_id"],
+            user=request.user,  # type: ignore
+            # only need to flag the removed dashboard to the log
+            after=lambda x: [],
+            before=lambda tile: [{"dashboard": {"id": tile.dashboard.id, "name": tile.dashboard.name}}],
+        )
+
+        return Response(status=status.HTTP_200_OK)
+
+    def _write_to_activity_log(
+        self,
+        insight_id: int,
+        dashboard_id: int,
+        user: User,
+        before: Optional[Callable[[DashboardTile], Optional[Any]]] = None,
+        after: Optional[Callable[[DashboardTile], Optional[Any]]] = None,
+    ) -> None:
+        try:
+            dashboard_tile = (
+                DashboardTile.objects.filter(
+                    insight_id=insight_id,
+                    dashboard_id=dashboard_id,
+                )
+                .select_related("dashboard", "insight")
+                .first()
+            )
+            if not dashboard_tile or not dashboard_tile.insight:
+                raise DashboardTile.DoesNotExist("the requested tile doesn't exist")
+            changes: List[Change] = [
+                Change(
+                    type="Insight",
+                    action="changed",
+                    field="dashboards",
+                    before=before(dashboard_tile) if before else None,
+                    after=after(dashboard_tile) if after else None,
+                )
+            ]
+            log_activity(
+                organization_id=self.organization_id,
+                team_id=self.team_id,
+                user=user,
+                item_id=insight_id,
+                scope="Insight",
+                activity="updated",
+                detail=Detail(
+                    name=dashboard_tile.insight.name
+                    if dashboard_tile.insight.name
+                    else dashboard_tile.insight.derived_name,
+                    changes=changes,
+                    short_id=dashboard_tile.insight.short_id,
+                ),
+            )
+        except Exception as e:
```

Comment:
Why handle this concern here vs in `log_activity`?

Reply:
Only to catch this https://github.com/PostHog/posthog/pull/13256/files#diff-b2afddf32d85b14129089203cc12f9d42aafba661c9122c17402fa83831aaf8cR137

So, if we can't load a tile the error is swallowed. Might be being overly protective

Reply:
Link only leads to whole commit - did you intend to link to a specific error?